### PR TITLE
Blog app: add API endpoints

### PR DESCRIPTION
**Pull request by brenoxav**
## Blog app: add API endpoints

- Create an API endpoint to list all posts.
- Create an API endpoint to list all comments for a post.
- API endpoints should receive JSON and respond JSON as well.

### CODE REVIEW

File: config/routes.rb [hunk 1]
```diff
@@ -10,7 +10,7 @@
   resources :users, only: [:index, :show]
   resources :friendships, only: [:index, :create, :update, :destroy]
   resources :posts, only: [:index, :create] do
-    resources :comments, only: [:create]
+    resources :comments, only: [:index, :create]
     resources :likes, only: [:create, :destroy]
   end
```
Comment (AshakaE) [inline config/routes.rb]: - [x] Kindly add endpoints for login and registration, I can only see index and show

**Review by AshakaE — CHANGES_REQUESTED** [review]
### Hi Team 🙋‍♀️, 

Good job so far! 👌

Pull request is 👌 
Git flow ✔️ 

There are some issues that you still need to work on to go to the next project but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>


#  Required Changes ♻️ 🚧
_Check the comments under the review._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: config/routes.rb (hunk 1 above)
Comment (StarSheriff2) [reply config/routes.rb]: We have added API authentication to the app. However, users can't sign up through the API, only sign in, so you will need to first create a user, before you are able to add a comment to a post, using the API.

**Comment by StarSheriff2** [comment]
Detailed instructions about how to use the API calls can be found in the README file. 😊

File: config/routes.rb (hunk 1 above)
Comment (vichuge) [reply config/routes.rb]: - [x] As my companion said, is important to create the endpoint for register, I kindly invite you to add it to your project. 🐶 
image

**Review by vichuge — CHANGES_REQUESTED** [review]
# Required Changes ♻️ ♻️ ⚠️ ⚠️ 🚦 🚦 

Hi dear student, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but some are still not addressed yet.

star_wars_almost_there


## Aspects to improve ♻️
_Check the comments under the review._

- [x] Check the comments above the review. 

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: config/routes.rb (hunk 1 above)
Comment (vichuge) [reply config/routes.rb]: Just do this and you'll finish this project. 🐱

Comment (StarSheriff2) [reply config/routes.rb]: Hi, we appreciate the feedback. Although this requirement was not clearly stated in the project description, we have implemented this feature. Users can now register through the Api.

**Comment by brenoxav** [comment]
Like Arturo said in the previous comment: 

> Although this requirement was not clearly stated in the project description, we have implemented this feature. Users can now register through the Api.

We added detailed instructions on how to use the API in the README.

**Review by emmanuellekamwa — APPROVED** [review]
**STATUS: APPROVED ✔🎉**
Hello @brenoxav 👋🏽
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

**Review by SpaYco — COMMENTED** [review]
Hello 👋,

_This is an additional review as a result of the second opinion request._

Thank you for bringing this to our attention, I'll make sure we update the project requirements so it's clear that signup is part of the project

Thank you for being professional throughout this project, I'm glad your project got approved

Keep rocking 🚀!

**Pull request merged**